Taskfile.yml

```diff
@@ -69,53 +73,111 @@ tasks:
     desc: Build stboot iso image
     deps:
       - check-config
+      - check-iso
       - boot-part
     sources:
       - "{{.BOOT_PART}}"
     generates:
       - "{{.ISO}}"
     cmds:
       - "rm -rf cache/iso"
-      - "mkdir -p cache/iso"
+      - "mkdir -p cache/iso; {{.CMD_COPY_HOST_CONFIG}}"
       - "cp {{.BOOT_PART}} cache/iso/boot_partition.vfat"
       - "xorriso -as mkisofs -R -f -e boot_partition.vfat -no-emul-boot -o {{.ISO}} cache/iso"
     preconditions:
-      - sh: '[ "{{.ST_BOOT_MODE}}" == "network" ]'
-        msg: "ISO image does only support network boot mode"
+      - sh: >-
+          [ "{{.ST_HOST_CONFIG_LOCATION}}" != "cdrom" ] ||
+          [ -n "{{.ST_HOST_CONFIG_FILE}}" ]
+        msg: |-
+          [ERROR] Configuration ST_HOST_CONFIG_FILE not defined in {{.CONFIG}}.
+      - sh: >-
+          [ "{{.ST_HOST_CONFIG_LOCATION}}" != "cdrom" ] ||
+          [ -f "{{.ST_HOST_CONFIG_FILE}}" ]
+        msg: |-
+          [ERROR] Host configuration file {{.ST_HOST_CONFIG_FILE}} missing.
     run: once
+    vars:
+      CMD_COPY_HOST_CONFIG:
+        sh: |
+          if [ "{{.ST_HOST_CONFIG_LOCATION}}" = "cdrom" ]; then
+          echo "cp {{.ST_HOST_CONFIG_FILE}} cache/iso/host_configuration.json"
+          fi
 
   disk:
     desc: Build stboot disk image
     deps:
       - check-config
+      - check-disk
       - boot-part
       - data-part
     sources:
       - "{{.BOOT_PART}}"
       - "{{.DATA_PART}}"
     generates:
-      - "{{.IMAGE}}"
+      - "{{.DISK}}"
     method: timestamp
     cmds:
-      - "{{.IMAGE_SCRIPT}} --boot {{.BOOT_PART}} --data {{.DATA_PART}} --output {{.IMAGE}}"
+      - "{{.IMAGE_SCRIPT}} --boot {{.BOOT_PART}} --data {{.DATA_PART}} --output {{.DISK}}"
     run: once
     vars:
       IMAGE_SCRIPT: ./scripts/image.sh
 
+  proviso:
+    desc: Build provisioning iso image
+    deps:
+      - check-config
+      - task: linux:kernel
+        vars:
+          IMAGE: "{{.PROVISIONING_LINUX_IMAGE}}"
+          INITRAMFS: "{{.PROVISIONING_INITRAMFS}}"
+          INITRAMFS_TARGET: "stboot:provisioning-initramfs"
+    sources:
+      - ./{{.PROVISIONING_LINUX_IMAGE}}
+    generates:
+      - ./{{.PROVISO}}
+    cmds:
+      - rm -rf {{.PROVISO_FAT}}
+      - mkdir -p {{.ISO_DIR}}
+      - mkfs.vfat -C -n PROVISO {{.PROVISO_FAT}} {{.BLOCKS}}
+      - mmd -i {{.PROVISO_FAT}} ::/EFI
+      - mmd -i {{.PROVISO_FAT}} ::/EFI/BOOT
+      - mcopy -i {{.PROVISO_FAT}} {{.PROVISIONING_LINUX_IMAGE}} ::/EFI/BOOT/BOOTX64.EFI
+      - xorriso -as mkisofs -R -f -e provision.fat -no-emul-boot -o {{.PROVISO}} {{.ISO_DIR}}
+    vars:
+      ISO_DIR: cache/proviso
+      PROVISO_FAT: ./{{.ISO_DIR}}/provision.fat
+      # TODO: Calculate size
```

Comment:
should something like `sh: du -BM {{.PROVISIONING_LINUX_IMAGE}}` work to get the size?

Reply:
Hmm, the issue is that it gets evaluated before the Linux image was build. Therefore, I was not able to apply the same calculation logic as done in the scripts. Let me solve this issue i a separate PR :+1:

Reply:
Sounds good